### ISSUE TITLE
Dynamic properties {valueOf} no longer work.

### ISSUE THREAD

**Issue by n17r4m** [pixijs/pixi.js#3468]
Sometime between v4.0.0 and v4.1.x dynamic properties no longer work. 

For example, in my game we've made plentiful use of:

`someSprite.position.x = {valueOf: function(){ return /* a + b + something */ }}`

We've found this to be a powerful way to construct a reactive system that can auto-compute layouts and positioning. However this no longer works in the latest versions. I'm guessing this is due to some caching of values for performance or similar, however it is preventing us from using the latest features and other performance improvements. Please consider reverting to direct property access to support this powerful paradigm.

**Comment by staff0rd** [comment]
Hi @n17r4m, are you able to supply a [pen](https://codepen.io) that displays this issue you're having against the latest [dev build](//d157l7jdn8e5sf.cloudfront.net/dev/pixi.js), and perhaps another against v4.0.0 that shows it working?

**Comment by n17r4m** [comment]
Sure. It appears as though the breaking change happened between 4.0.0-rc3 and 4.0.0-rc4

Working: http://codepen.io/n17r4m/pen/BQGZqv
Not Working: http://codepen.io/n17r4m/pen/yVQoLe

**Comment by staff0rd** [comment]
Your `valueOf` assignment actually still works, its just that the transform is no longer being updated every frame (for performance reasons).

You could however force the transform to be updated like I did [here](http://codepen.io/staff0rd/pen/oYQeoa) by doing this in your render loop;

~~~
circ.transform.onChange();
~~~

**Comment by n17r4m** [comment]
I understand, however this is our primary method of updating positions, rotations, etc on most of our display objects. I feel that being able to reactively declare functional targets for display updates is important, and this has streamlined and modularized the code base. 

Could you recommend a method to monkey patch in the previously available functionality, such that on each frame render the currently specified translations and transformations are queried and applied? I noticed that in codepen the animation was smoother with the latest version of PIXI, despite the transform.onChange. Presently, all x,y positions and scale parameters in our unreleased http://fly.lampop.com game are specified using this valueOf technique. 

Perhaps there some class method in the PIXI hierarchy that we can add functional advice to so as to call .transform.onChange() on every display object on every frame?

**Comment by staff0rd** [comment]
If you check your working pen, you'll see that the reason it's working is because `this._localID` and `this._currentLocalID` are both `NaN` in [this line](https://github.com/pixijs/pixi.js/blob/6c617a5330a5fbee43d354ba5d7150db5ea1830f/src/core/display/TransformStatic.js#L117) (which seems like a bug), which is why I called `onChange` to ensure the conditonal is always met.

Any monkey patch would just need to ensure entry to that block.

**Comment by n17r4m** [comment]
Ok, I've refactored our code base so that per-frame property updates are no longer handled by {valueOf} magic, but since all of the layout is heavily dependant on this, I wrote a small recursive update utility that I can apply to the root scene display object. 

Maybe someone else will find it useful.

```
function recursiveTransformUpdate(dispOb){
    var ref$, ref1$;
    if ((ref$ = dispOb.transform) != null && ref$.onChange) {
      dispOb.transform.onChange();
    }
    if ((ref1$ = dispOb.children) != null && ref1$.length) {
      return dispOb.children.forEach(recursiveTransformUpdate);
    }
};
```

Marking issue as closed. I think having some test for {valueOf} properties and having .onChange() be called automatically on render would be nice, however I understand if you feel it may not be worth it to support this corner case.
